### PR TITLE
Prac 5 code review request

### PR DESCRIPTION
**Pull request by DanL-Smith**
Please check formatting, naming and logic.
@codysullivan1

### CODE REVIEW

**Comment by lindsaymarkward** [comment]
You should only have one "Initial Commit" per repo, not per folder/file.

File: prac_05/emails.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+email_to_name = {}
+email = input("Email: ")
+while email != "":
+    modified_email = email.replace(",", " ").replace(".", " ")
+    name = " ".join([char for char in (modified_email[:modified_email.find("@")].split()) if not char.isdigit()]).title()
```
Comment (lindsaymarkward) [inline prac_05/emails.py:5]: This is not very readable. It's not usually better to write hard-to-read code on one line.

File: prac_05/emails.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    confirm = input(f"Is your name {name}? (Y/n)").lower()
+    if confirm == 'y' or "":
```
Comment (lindsaymarkward) [inline prac_05/emails.py:7]: Why use the "no else" pattern here?
`or ""` and `'n' or 'no'` don't do what you (probably) think they do.
https://github.com/CP1401/Practicals/tree/master/prac_03#note

File: prac_05/emails.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        email_to_name[email] = name
+    elif confirm == 'n' or 'no':
+        name = input('Name: ')
+        email_to_name[email] = name
```
Comment (lindsaymarkward) [inline prac_05/emails.py:11]: DRY

File: prac_05/state_names.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+print(CODE_TO_NAME)
+
+state_code = input("Enter short state: ").upper()
+while state_code != "":
+    if state_code in CODE_TO_NAME:
+        print(state_code, "is", CODE_TO_NAME[state_code])
+    else:
+        print("Invalid short state")
+    state_code = input("Enter short state: ").upper()
+for key, value in CODE_TO_NAME.items():
```
Comment (lindsaymarkward) [inline prac_05/state_names.py:19]: Don't use generic meaningless names like key (which is what in this case?)

File: prac_05/word_occurrences.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+"""
+CP1404/CP5632 Practical
+Word Occurrences
+"""
+from collections import OrderedDict
```
Comment (lindsaymarkward) [inline prac_05/word_occurrences.py:5]: Why?

File: prac_05/emails.py (hunk 1 above)
Comment (DanL-Smith) [reply prac_05/emails.py:5]: I have moved this section in to a get_names_from_email function and changed the code to be more readable.

File: prac_05/emails.py (hunk 2 above)
Comment (DanL-Smith) [reply prac_05/emails.py:7]: I see my error here. Thank you.

File: prac_05/emails.py (hunk 3 above)
Comment (DanL-Smith) [reply prac_05/emails.py:11]: Yes, a very important principle. Thank you for pointing this out.

File: prac_05/state_names.py (hunk 4 above)
Comment (DanL-Smith) [reply prac_05/state_names.py:19]: Yep this should definitely be less ambiguous. Thank you.

File: prac_05/word_occurrences.py (hunk 5 above)
Comment (DanL-Smith) [reply prac_05/word_occurrences.py:5]: This has now been removed. :)

**Pull request merged**